Still having difficulties in the saving function(save.py). Will leave it for now.